tests/model_fields/tests.py

```diff
@@ -271,6 +271,14 @@ class Meta:
 
         self.assertEqual(ConcreteReferent._meta.get_field('reference').related_model, Refered)
 
+    @test.skipUnlessDBFeature('can_defer_constraint_checks')
+    def test_defers_constraint_check(self):
+        b = Bar.objects.create(b='abc', a_id=5)
+        self.assertEqual(b.a_id, 5)
+        # Delete the object to avoid the error once constraints are
+        # checked.
+        b.delete()
```

Comment:
Is there precedent for adding tests for the `TestCase` class? I couldn't find any to use as a starting point. I'm not sure how to go about adding a regression test for this change.